### ISSUE TITLE
Read default application.yaml configmap

### ISSUE THREAD

**Issue by shahbour** [spring-cloud/spring-cloud-kubernetes#153]
Coming from `spring-cloud-config`, and while converting my application into `spring-cloud-kuberentes-config`.

I found that having a common configuration file for your application help (it can hold database connection configuration, Redis connection, RabbitMQ .... ) so all common configuration can be in one place. 

Another missing feature is having config map per profile 

For example, if we a spring boot application name  called demo with production profile active then we look for config map with name 

`application`
`demo`
`demo-production` 

currently only below config map is fetched

`demo`

Is adding this flexibility on the roadmap, if yes I can start preparing a pull request

**Comment by spencergibb** [comment]
I thought it already did that. @iocanel, @geoand or @salaboy care to comment?

If not, it should be considered.

**Comment by shahbour** [comment]
Currently, multiple config maps are supported, but there is no link between Active Profiles and Config Map Names. In case we created multiple config maps per profile they have to be added manually to the `spring.application.kubernetes.config.sources`.

**Comment by geoand** [comment]
@shahbour @spencergibb I was also under the impression that we did that.
If not, then I agree it should definitely be considered.

@shahbour Do you have a sample I could try?

**Comment by spencergibb** [comment]
ping @shahbour

**Comment by shahbour** [comment]
@spencergibb @geoand , just to be clear we are talking about checking configmaps based on names provided in `spring.application.kubernetes.config.sources` and active profiles .  If this is the case then yes it is not supported we have to add the configmap name with profile in bootstrap.yml . for example if we have
```
spring:
  profiles:
    active: production
  application:
    name: cdr-import
  cloud:
    kubernetes:
      config:
        namespace: default
        sources:
         - name: global
         - name: cdr-import
```

It will only check for configmap `global` and `cdr-import` and not `global-production` and `cdr-import-production` . 

I will provide a sample tomorrow if needed , in my case i had to manually add `cdr-improt-produciton` to the list

**Comment by spencergibb** [comment]
OK, so we do this, but not with `*.config.sources` is what I am seeing. @geoand agree?

**Comment by spencergibb** [comment]
There's an open PR for profiles support https://github.com/spring-cloud/spring-cloud-kubernetes/pull/323

**Comment by geoand** [comment]
@spencergibb @shahbour yeah, it makes perfect sense

**Comment by shahbour** [comment]
I tried reading PR #323 and I think it is the same as now for reading one config map based on the name provided without postfix any active profile name.

from `ConfigMapPropertySource`

```
	private static Map<String, String> getData(KubernetesClient client, String name,
			String namespace, Environment environment) {
		try {
			ConfigMap map = StringUtils.isEmpty(namespace)
					? client.configMaps().withName(name).get()
					: client.configMaps().inNamespace(namespace).withName(name).get();

			if (map != null) {
				return processAllEntries(map.getData(), environment);
			}
		}
		catch (Exception e) {
			LOG.warn("Can't read configMap with name: [" + name + "] in namespace:["
					+ namespace + "]. Ignoring");
		}

		return new HashMap<>();
}
```

I think we should change the issue name because reading default configuration is already done , what is missing is reading multiple config-map based on active profiles.

**Comment by geoand** [comment]
@shahbour perhaps open a new issue where you set the appropriate title and link the discussion here?

**Comment by shahbour** [comment]
This issue was opened before providing the `spring.application.kubernetes.config.sources` which did address partially this issue. Personally, I think the current solution is good unless you think the same behaviour of active profiles and yml files should be mocked into configmap .

**Comment by geoand** [comment]
I can see reasons for doing both to tell you the truth, so I would like to hear more opinions

**Comment by shahbour** [comment]
I see, ok I have worked on this before in pr #162  , after #323 is merged I can work on it again?

**Comment by geoand** [comment]
Sounds good to me :)

**Comment by salaboy** [comment]
@geoand @shahbour sorry for my lack of participation here, but I totally agree.. having the profile information while reading configmaps will be really useful

**Comment by ryanjbaxter** [comment]
Fixed via 296bf1e
